data/opam_docs/2.1/03_faq.md

```diff
@@ -0,0 +1,446 @@
+* Check for pending reinstallations `opam reinstall --list-pending`
+
+---
+
+#### 🐫  When trying to install a new package, opam wants to remove or downgrade packages that I have installed. How to know why?
+
+There is likely a conflict between them or their dependencies and what you are
+requesting, here is how to find out. We'll suppose you were trying to install
+`foo` and `bar` got removed:
+
+* `opam install foo bar`, if not possible, will tell you why.
+* The above may find a solution by using older version of the packages, in that
+  case try and force the latest versions thusly: `opam install foo.2.0 bar.1.1`
+  (you can also use constraints like `'foo>=2.0'`).
+
+---
+
+#### 🐫  Where do I report Bugs, Issues and Feature Requests?
+
+- Bug reports and feature requests for the opam **tool** should be reported on
+[opam's issue-tracker](https://github.com/ocaml/opam/issues). Please include the
+output of `opam config report` whenever applicable.
+
+- Packaging issues or requests for a new package can be reported on the
+[official repository's
+issue-tracker](https://github.com/ocaml/opam-repository/issues).
+
+- General queries for both the tool and the packages can be addressed on the
+[OCaml-platform mailing-list](http://lists.ocaml.org/listinfo/platform) and
+insights and evolution of opam internals can discussed on the [opam-devel
+mailing-list](http://lists.ocaml.org/listinfo/opam-devel).
+
+- https://discuss.ocaml.org is a good place for community assistance.
+
+- You may also try IRC channel `#ocaml` on Libera.Chat.
+
+---
+
+#### 🐫  How to link to libraries installed with opam?
+
+The standard way of doing this is to use
+[ocamlfind](https://opam.ocaml.org/packages/ocamlfind), which is orthogonal to
+opam: `ocamlfind query <lib>`. If you use [dune](https://github.com/ocaml/dune),
+this should be completely transparent.
+
+Your libraries are installed to the directory returned by ``opam var lib``,
+which is by default `~/.opam/<switch>/lib`. Note that using `ocamlc`'s option
+`-I +dir` will make `dir` relative to `lib/ocaml`, and will only work for the
+libraries that ship with the compiler. Also, remember to add the dependency when
+you package your project!
+
+---
+
+#### 🐫  How does opam tell which version of a package is newer?
+
+We use the basics of the [version
+ordering](https://www.debian.org/doc/debian-policy/ch-controlfields.html#s-f-Version)
+from Debian:
+
+- version strings are sliced into alternate, possibly empty non-digit / digit
+  sequences, always starting with a non-digit sequence.
+- those are sorted lexicographically, using resp. ASCII (with symbol > letter)
+  and number order. For example `a` gives `["a"]`, and `1` gives `["";1]`, so
+  `a` is latest (`"" < "a"`).
+- the `~` character is special as it sorts even before the end of sequence. It's
+  most convenient for pre-releases, allowing `1.0~beta` to be before `1.0`.
```

Comment:
```suggestion
- Those are sorted lexicographically, using resp. ASCII (with symbol > letter)
  and number order. For example `a` gives `["a"]` and `1` gives `["";1]`, so
  `a` is latest (`"" < "a"`).
- The `~` character is special because it sorts even before the end of sequence. It's
  most convenient for pre-releases, allowing `1.0~beta` to be before `1.0`.
```